dataframe/typed_array_builder.go

```diff
@@ -0,0 +1,197 @@
+package dataframe
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+type typedArrayBuilder struct {
+	size       int
+	keyList    []string
+	valInts    []int
+	valFloats  []float64
+	valObjs    []string
+	whichVals  int
+	dType      string
+	currType   string
+	buildError error
+}
+
+// A Series contains values of one of these three types. The which field uses these
+// constants to determine which slice holds the actual values of the Series.
+const (
+	typeInt   = 1
+	typeFloat = 2
+	typeObj   = 3
+)
+
+func newTypedArrayBuilder(size int) *typedArrayBuilder {
+	return &typedArrayBuilder{
+		size: size,
+	}
+}
+
+func (t *typedArrayBuilder) setType(dtype string) {
+	if dtype == "" {
+		return
+	}
+	t.dType = dtype
+	t.currType = dtype
+	if t.dType == "int64" || t.dType == "bool" {
```

Comment:
These `dtype` strings are starting to make more sense to me now as someone who isn't super familiar with the underpinnings of pandas. These `dtype`s are coming from `ndarray`, which is implemented in C. So up in python land `int64` isn't a thing, but it's a window down into "stronger typed array land". Not surprised to see these being used aggressively here. To that end, we may want to do the same thing ndarray is doing, and work toward matching the actual type of the go slice to the reported type.

Good god this will benefit from generics.